pkg/scanners/factory.go

```diff
@@ -41,5 +42,8 @@ func NewFactory(set registries.Set) Factory {
 	quayScannerType, quayScannerCreator := quayScanner.Creator()
 	reg.creators[quayScannerType] = quayScannerCreator
 
+	scannerV4Type, scannerV4Creator := scannerv4.Creator(set)
+	reg.creators[scannerV4Type] = scannerV4Creator
```

Comment:
Could surround it by a feature flag, should not be necessary based on the design, but suppose it will not hurt. Added a note to ROX-21039, thats the story introducing the feature flag in other areas of the implementation.

A delayed integration I don't believe provides value based on the current design - also refer to ROX-21039 (and the design) for the when we'll be creating / deleting the integration. It is not in scope for this story.  

The Scanner V2 delayed integration only has benefit on a fresh install of ACS, any subsequent startup it doesn't add value. Since 'intent' to use Scanner V4 is driven by the feature flag we'll create the integration when we see the feature flag (based on current design, that may change if we make the feature opt in, at which point it still wouldn't be delayed and would be driven by creating an integration manually).

Reply:
I think the main benefit for the delayed integration is that we don't try to make the integration available until we know Scanner is available. I think if we make this a normal integration, it is assumed Scanner v4 is already alive and ready, which may not be the case, as Central may be ready before Scanner v4 is. I'm under the impression we won't bother trying to scan anything until the integration is made, which is why I think this would be beneficial (but I could be wrong about Central waiting for the integration to be available before trying to scan)

Reply:
> I'm under the impression we won't bother trying to scan anything until the integration is made
It's true that nothing will be scanned by the initial reprocess loop if no scanners are avail (and the log message suppressed).  However this only applies to the very first / fresh install of ACS.  Any subsequent upgrades or Central restarts the logs will be spammed with errors because an integration will exist (the delayed integration no longer has an effect).

At this point I consider this a nice to have to reduce initial log spam - added ROX-21434 for future consideration.